datafusion/common/src/scalar.rs

```diff
@@ -2937,13 +3015,33 @@ impl fmt::Display for ScalarValue {
                 )?,
                 None => write!(f, "NULL")?,
             },
-            ScalarValue::List(arr)
-            | ScalarValue::LargeList(arr)
-            | ScalarValue::FixedSizeList(arr) => {
+            ScalarValue::List(arr) => {
                 // ScalarValue List should always have a single element
                 assert_eq!(arr.len(), 1);
                 let options = FormatOptions::default().with_display_error(true);
-                let formatter = ArrayFormatter::try_new(arr, &options).unwrap();
+                let formatter =
```

Comment:
The same pattern could be applied here as in https://github.com/rspears74/arrow-datafusion/pull/1 (use `first_array_for_list`)

Reply:
I'll take care of this one too. I went through my changes looking for any larger duplication sections but missed this one.